### PR TITLE
Test for issue #7454: fmt triggers menhir

### PR DESCRIPTION
**Pull request by voodoos**
See issue #7454.

## Expected Behavior
I would expect `dune fmt` or `dune build @fmt` to never build anything.
(This is an assumption made by OCaml-CI for the `lint-fmt` job. cc @maiste)

## Actual Behavior
Missing menhir parsers that have a corresponding `.mli` file are generated when calling `dune build @fmt`

### CODE REVIEW

File: test/blackbox-tests/test-cases/formatting/no-gen.t [hunk 1]
```diff
@@ -0,0 +1,78 @@
+See issue 7454
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/formatting/no-gen.t:1]: Could you paste the relevant details from the issue here?

**Review by rgrinberg — APPROVED** [review]
I guess the issue is that we're setting up formatting rules for all sources used for building. We should only be formatting sources that are present in the sources.

**Comment by anmonteiro** [comment]
I wonder if https://github.com/ocaml/dune/pull/6585 is related to this.

**Review by emillon — APPROVED** [review]
Sounds good to me too. Please add DCO and I'll rebase+merge.

**Comment by voodoos** [comment]
> Sounds good to me too. Please add DCO and I'll rebase+merge.

Done !

**Comment by rgrinberg** [comment]
None of the uses in this PR require the `-a` flag either. So using `ls` without flags should be fine too

**Pull request merged**